python/pyarrow/src/arrow/python/helpers.cc

```diff
@@ -292,7 +292,11 @@ bool PyFloat_IsNaN(PyObject* obj) {
 
 namespace {
 
+#ifdef Py_GIL_DISABLED
```

Comment:
No need for conditional compilation IMHO, let's just use `std::once_flag` always.

Reply:
I used a conditional cause this could theoretically introduce a deadlock when the GIL is there. You removed the `std::once_flag` in https://github.com/apache/arrow/commit/f69061935e92e36e25bb891177ca8bc4f463b272.

Reply:
Aha, good catch. Can you add a comment then?

Reply:
Comment added.